lib/puppet/defaults.rb

```diff
@@ -257,8 +257,9 @@ def self.valid_file_checksum_types
           on the CLI.",
     },
     :configprint => {
-        :default  => "",
-        :desc     => "Prints the value of a specific configuration setting.  If the name of a
+        :default    => "",
+        :deprecated => :completely,
+        :desc       => "Prints the value of a specific configuration setting.  If the name of a
```

Comment:
Ugh, I caught this one while skimming the file and added it. I suppose it should be in a different PR, or at least different commit though?

Reply:
The `--configprint` option/setting applies to all subcommands, e.g.

```
$ bundle exec puppet apply --configprint server
localhost
```

But it is deprecated (with `puppet config` being the replacement), and I'm fine deprecating `configprint` in this PR.